cmd/zetae2e/local/bitcoin.go

```diff
@@ -5,71 +5,147 @@ import (
 	"time"
 
 	"github.com/fatih/color"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/zeta-chain/node/e2e/config"
 	"github.com/zeta-chain/node/e2e/e2etests"
 	"github.com/zeta-chain/node/e2e/runner"
+	"github.com/zeta-chain/node/testutil"
 )
 
-// bitcoinTestRoutine runs Bitcoin related e2e tests
-func bitcoinTestRoutine(
+// startBitcoinTestRoutines starts Bitcoin deposit and withdraw tests in parallel
+func startBitcoinTestRoutines(
+	eg *errgroup.Group,
 	conf config.Config,
 	deployerRunner *runner.E2ERunner,
 	verbose bool,
-	initBitcoinNetwork bool,
-	testNames ...string,
-) func() error {
-	return func() (err error) {
-		account := conf.AdditionalAccounts.UserBitcoin
-		// initialize runner for bitcoin test
-		bitcoinRunner, err := initTestRunner(
-			"bitcoin",
-			conf,
-			deployerRunner,
-			account,
-			runner.NewLogger(verbose, color.FgYellow, "bitcoin"),
-		)
-		if err != nil {
-			return err
-		}
+	initNetwork bool,
+	depositTests []string,
+	withdrawTests []string,
+) {
+	// initialize runner for deposit tests
+	runnerDeposit := initRunnerDeposit(conf, deployerRunner, verbose, initNetwork)
 
-		bitcoinRunner.Logger.Print("🏃 starting Bitcoin tests")
-		startTime := time.Now()
+	// initialize runner for withdraw tests
+	runnerWithdraw := initRunnerWithdraw(conf, deployerRunner, verbose, initNetwork)
+
+	// initialize funds
+	// send BTC to TSS for gas fees and to tester ZEVM address
+	if initNetwork {
+		// mine 101 blocks to ensure the BTC rewards are spendable
+		// Note: the block rewards can be sent to any address in here
+		_, err := runnerDeposit.GenerateToAddressIfLocalBitcoin(101, runnerDeposit.BTCDeployerAddress)
+		require.NoError(runnerDeposit, err)
+
+		// send BTC to ZEVM addresses
+		runnerDeposit.DepositBTC(runnerDeposit.EVMAddress())
+		runnerDeposit.DepositBTC(runnerWithdraw.EVMAddress())
+	}
```

Comment:
I don't understand why we can't do this in the individual runners?

Reply:
I think this makes sense, but we should modify `DepositBTC` to prevent race condition

Reply:
This might also be faster for the runnerDeposit to just send the BTC to the two accounts directly on Bitcoin, and the runners manage their own tests

Reply:
Yeah. As @lumtis said, preventing race condition is one of the reasons. Other issues were

1. only the deposit runner holds funds in UTXO, it then sends test BTC token to both runners. There was a limitation of one single node wallet here, the zetaclients UTXO query will fail if we create two wallets.
2. if we generate `101` twice, zetaclient will take quite a bit to catch up blocks.